packages/router/router.ts

```diff
@@ -1039,17 +1041,27 @@ export function createRouter(init: RouterInit): Router {
   // Trigger a navigation event, which can either be a numerical POP or a PUSH
   // replace with an optional submission
   async function navigate(
-    to: number | To,
+    to: number | To | null,
     opts?: RouterNavigateOptions
   ): Promise<void> {
     if (typeof to === "number") {
       init.history.go(to);
       return;
     }
 
-    let { path, submission, error } = normalizeNavigateOptions(
+    let normalizedPath = normalizeTo(
+      manifest,
+      state.location,
+      state.matches,
+      basename,
+      future.v7_prependBasename,
       to,
-      future,
+      opts
+    );
+    let { path, submission, error } = normalizeNavigateOptions(
+      future.v7_normalizeFormMethod,
+      false,
+      normalizedPath,
       opts
     );
```

Comment:
Normalizing the path (`normalizeTo`) is now separate from normalizing the submission info (`normalizeNavigateOptions`).